src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx

```diff
@@ -229,6 +250,18 @@ export class SavedObjectsTable extends Component<SavedObjectsTableProps, SavedOb
       findOptions.sortField = 'type';
     }
 
+    if (taggingApi && selectedTags) {
+      const references: SavedObjectsFindOptionsReference[] = [];
+      selectedTags.forEach((tagName) => {
+        const ref = taggingApi.ui.convertNameToReference(tagName);
+        if (ref) {
+          references.push(ref);
+        }
+      });
+      findOptions.hasReference = references;
+      findOptions.hasReferenceOperator = 'OR';
```

Comment:
do see it's tested. can we probably extract that logic out of UI components to simplify testing?